crates/telio-lana/src/moose_callbacks.rs

```diff
@@ -1,19 +1,28 @@
 pub use crate::event_log::*;
+use std::sync::mpsc::SyncSender;
 pub use telio_utils::{telio_log_error, telio_log_info, telio_log_warn};
 
-/// Struct for Moose callbacks
-pub struct MooseCallbacks;
+/// Struct for moose::InitCallback
+pub struct MooseInitCallback {
+    /// Channel tx half to send init result
+    pub init_tx: SyncSender<Result<moose::TrackerState, moose::MooseError>>,
```

Comment:
Moose side implementation is expecting a blocking send call, and besides that we set the the channel buffer to 1, is there any extra reason to choose oneshot over the actual `SyncSender`?

> Also, we don't really read the value inside the result, so maybe we only need to send a bool and rename the field to something like init_success.

Agreed, will change.

Reply:
The main reason would be to express the fact that the is not an ordinary channel but only a single value will ever be transmitted over it. When someone reads this in future it will be obvious from the type that this will never be used to send multiple value. Like I said, it's a nitpick :)